src/analyzers/dingo/testdata/src/proper_inject_tags/test_proper_inject_tags.go

```diff
@@ -32,19 +42,33 @@ type Z struct {
 	X bool `inject:""` // want `Empty Inject-Tags are not allowed! Add more specific naming or use the Inject function for non configuration injections`
 }
 
+func providerFunc(e *E) interface{} {
+	return new(interface{})
+}
+
+func (d *D) providerFuncWithSelector(e *E, f *F) interface{} {
+	return new(interface{})
+}
+
 func (d *D) Inject(
 	service *A,
 	z *Z,
 	cfg *specialInjectCfg,
 	annotated *struct {
-		Mapper Mapper `inject:"my-annotations,optional"` // this is allowed as it is an argument of the Inject function
-	},
+	Mapper Mapper `inject:"my-annotations,optional"` // this is allowed as it is an argument of the Inject function
+},
 ) *D {
 
-	d.servie = service
+	d.service = service
 	if cfg != nil {
 		d.isDebug = cfg.IsDebug
 	}
 
 	return d
 }
+
+func (d *D) Configure(injector *dingo.Injector) bool {
+	injector.Bind(new(interface{})).ToProvider(providerFunc)
+	injector.BindMulti(new(interface{})).ToProvider(d.providerFuncWithSelector)
```

Comment:
Let's also add a `BindMap` here for completeness.

Reply:
What about split declarations like 
```go
bind := injector.Bind(new(interface{}))
bind.ToProvider(providerFunc)
```
?

Reply:
That's a good point and something we doesn't check rn. Same for the binding_implements_interface analyzer, would be something I can jump on next